[aggregator] Add support for service checks from checks (part 1)

olivielpeau opened the pull request:
### What does this PR do?

Adds support for service checks for the go checks, and serialization to the `v1/check_run` endpoint.

Doesn't include:
- the python bindings for the py checks (will follow in an upcoming PR)
- the v2 endpoint support (I'll create a separate card for that)

### Testing Guidelines

Some basic tests. Also tested this manually.

The default service check `datadog.agent.up` does show up on DD.

### Additional Notes

See commit message for details. Easier to review by reviewing each commit one after the other.

The aggregator listens on a dedicated channel for service checks and stores them. It flushes them at the same time as the metrics (no specific reason to do that, except that it's slightly simpler to implement).

I also sneaked in a cosmetic change to the python check api implementation (renaming `submit_data` to `submit_metric`), so that adding a `submit_service_check` method (later) makes sense.

olivielpeau committed:
[aggregator] Add support for service checks from checks

The aggregator sends a `datadog.agent.up` service check by default
(like what the Agent5 collector does).

No aggregation is made on the service checks, so they're stored
directly in the BufferedAggregator.

For now:
- Haven't added the python bindings for py checks yet
- It only supports the v1 api endpoint (I'll queue some work to support
the v2 endpoint)

olivielpeau committed:
[py] Rename `submit_data` to `submit_metric`

masci reviewed (changes requested):
Only one comment to improve the diff while waiting for the part2, otherwise LGTM!

olivielpeau committed:
[aggregator] Rename check-metric-related types and variables

So that there's a clear distinction between what's made for metrics
and what's made for service checks

olivielpeau committed:
[aggregator] Serialize and send to forwarder in separate goroutines
